store/structured/mongo/store.go

```diff
@@ -0,0 +1,131 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	mongoDriver "go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+
+	"github.com/tidepool-org/platform/errors"
+)
+
+type Store struct {
+	client *mongoDriver.Client
+	config *Config
+}
+
+type Status struct {
+	Ping string
+}
+
+func NewStore(c *Config) (*Store, error) {
+	if c == nil {
+		return nil, errors.New("database config is empty")
+	}
+
+	store := &Store{
+		config: c,
+	}
+
+	cs := c.AsConnectionString()
+	clientOptions := options.Client().
+		ApplyURI(cs).
+		SetConnectTimeout(store.config.Timeout).
+		SetServerSelectionTimeout(store.config.Timeout)
+
+	var attempts int64 = 1
+	var err error
+	var closingChannel = make(chan bool, 1)
+	for {
+		var timer <-chan time.Time
+		if attempts == int64(0) {
+			timer = time.After(0)
+		} else {
+			timer = time.After(store.config.WaitConnectionInterval)
+		}
+		select {
+		case <-closingChannel:
+			close(closingChannel)
+			if err != nil {
+				return nil, err
+			}
+			return store, nil
+		case <-timer:
+			err = initConnexion(store, clientOptions)
+			if err == nil {
+				closingChannel <- true
+			} else {
+				if store.config.MaxConnectionAttempts > 0 && store.config.MaxConnectionAttempts <= attempts {
+					closingChannel <- true
+				} else {
+					attempts++
+				}
+			}
+		}
+	}
```

Comment:
As far as I can see, the NewStore function is blocking until we get a mongo connection. It's not the behavoiour we had until now, service should start even if mongo store is not available.